source/admin_manual/event_reasons.rst

```diff
@@ -0,0 +1,135 @@
+.. _event_reasons:
+
+=============
+Event Reasons
+=============
+
+.. versionadded:: v2.3.15
+
+List of "reason_code" values that used by :ref:`events <list_of_events>`.
+The reason codes are always in <module>:<name> format.
+
+Storage
+-------
+
+ * ``storage:autoexpunge`` - Mails are being autoexpunged
+ * ``mdbox:rebuild`` - mdbox storage is being rebuilt
+
+Mailbox
+-------
+
+ * ``mailbox:search`` - Mails are being accessed due to search query
+ * ``mailbox:sort`` - Mails are being sorted (IMAP SORT)
+ * ``mailbox:thread`` - Threading is being built for mails (IMAP THREAD)
+ * ``mailbox:vsize`` - mailbox vsize is asked
+ * ``mailbox:vsize_update`` - mailbox vsize is updated (append/expunge)
```

Comment:
Maybe just merge this into vsize? It almost never happens anyway and can't be easily tested..